runtime/regexp.cpp

```diff
@@ -706,6 +720,12 @@ Optional<int64_t> regexp::match(const string &subject, mixed &matches, bool all_
   pcre_last_error = 0;
 
   int64_t result = 0;
+  offset = initial_offset(subject, offset);
```

Comment:
Please, consider using `get_correct_index` instead of `initial_offset`
Something like
```cpp
offset = std::max(subject.get_correct_index(offset), 0_i64)
```

Reply:
Replaced initial_offset with your suggestion.
Nice one. :+1: